### PR TITLE
Add Alembic and package and file size

### PR DESCRIPTION
**Pull request by lxylxy123456**
1. Add Alembic
2. Make a python package so that it can be installed using pip
3. [edited] Add a file size column (just for management)

@rkevin-arch Please review and test carefully. The changes are likely to break setup.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import setuptools
+
+with open("README.md", "r") as fh:
+    long_description = fh.read()
+
+setuptools.setup(
+    name="ngshare",
+    version="0.0.1",
```
Comment (moorepants) [inline setup.py:8]: I usually keep a `ngshare/version.py` that only contains a single variable `__version__` containing a string of the version number. Then here you actually `exec(open('ngshare/version.py').read())` to make it available in setup.py. This gives a single source for the version number (can be used in sphinx docs too). You'll need to `from version import __version__` in your main `__init__.py` also, so `from ngshare import __version__` works.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    author="Team KALE",
+    author_email="ercli@ucdavis.edu",
+    description="nbgrader sharing service",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/pypa/sampleproject",
```
Comment (moorepants) [inline setup.py:14]: Point to this repo.

**Comment by moorepants** [comment]
What is alembic?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    packages=setuptools.find_packages(),
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        "License :: OSI Approved :: BSD License",
+        "Operating System :: POSIX :: Linux",
+    ],
+    python_requires='>=3.6',
+)
```
Comment (moorepants) [inline setup.py:22]: You need an `install_requires` keyword to setup the dependency constraints.

**Comment by lxylxy123456** [comment]
@moorepants Alembic is migration manager for SQLAlchemy. For example if I add a row to the database, Alembic can automatically update the SQLite database's schema.

**Comment by moorepants** [comment]
Thanks. Now that you say that, I think I've used it or was aware of it. I wrote an app that used sqlalchemy some years ago.

File: setup.py (hunk 1 above)
Comment (lxylxy123456) [reply setup.py:8]: I will look into the setup.py configuration later, but I am a little bit concerned whether it is a good practice to use `exec` or `eval`. When learning Python I was told that it is dangerous.

Comment (moorepants) [reply setup.py:8]: In general you should be very careful using exec and eval, but this is one case where I do use it. There is a chicken and egg problem if you want to store the version number in one location for use in the imported package and during installation, i.e. you need to run setup.py before ngshare is installed. This is a common practice in python setup.py files at this point and I use it on all of the 20+ packages I maintain.

Comment (moorepants) [reply setup.py:8]: See the official recommendations here: https://packaging.python.org/guides/single-sourcing-package-version/

Comment (lxylxy123456) [reply setup.py]: I am using the first solution (`get_version` function). It should work in most cases (as long as the version name is not a strange string).

**Comment by lxylxy123456** [comment]
Note: new reviews are recommended to be made in #67

**Comment by lxylxy123456** [comment]
Closed because will merge in `sudo` branch in #67

**Pull request closed without merging**